File: Median of BST.java
class Tree
{
    public static void inoder(Node root, ArrayList<Integer> list){
        if (root== null){
            return ;
        }
        inoder(root.left, list);
        list.add(root.data);
        inoder(root.right, list);
    }
    
    public static float findMedian(Node root)
    {
        ArrayList<Integer> list = new ArrayList<>();
        
        inoder(root, list);
        int size = list.size();
        
        if (size%2==0){
            int n = size/2;
            
            return ((float) (list.get(n) + list.get(n-1)))/2;
        }
        else 
        return list.get(size/2);
    }
}

File: Median of BST.cpp
vector<float> arr;
void inorder(Node* root) 
{
    if (!root) 
        return;
    
    inorder(root->left);
    arr.push_back(root->data);
    inorder(root->right);
}

float findMedian(struct Node* root) 
{
    
    arr.clear();
    inorder(root);
    int n = arr.size();
    
    if (n % 2==1)
        return arr[n/2];
    else
        return (arr[n/2 - 1] + arr[n/2])/2;
}
